### PR TITLE
LF-3872-not-able-to-edit-expense-value

### PR DESCRIPTION
**Pull request by navDhammu**
**Description**

This appears to be only reproducible in certain situations. 

**Editing an expense value that is between 1-9**

To reproduce:
1. Add an expense between $1 - $9 and save.
2. Try to edit the expense by pressing backspace, then entering a new value. The save button will remain disabled not allowing you to submit form. However, if you blur the input field after clearing it before entering a new value then it still works.

**Editing any expense value by first clearing the input at once**

To reproduce:
1. Add an expense with more than 1 digit and save.
2. Clear the input by highlighting the entire number at once and delete. 
3. Try to edit right away without blurring, you will find the save button remains disabled.

What seems to be happening is that when the value becomes empty, it is being set as NaN and for some reason that doesn't trigger the dirty or valid state of the form which doesn't trigger a rerender and the button remains disabled. It works in other cases because the dirty state is changed first before the value becomes empty (or NaN). 

Jira link: https://lite-farm.atlassian.net/browse/LF-3872

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/components/Finances/PureExpenseDetail/index.jsx
Comment (navDhammu) [inline packages/webapp/src/components/Finances/PureExpenseDetail/index.jsx:1]: Just replaced valueAsNumber with setValueAs and ensuring it never becomes NaN by setting empty value to null

**Review by Duncan-Brain — APPROVED** [review]
Looks good and works for me! 

It is also triggering validation which is great. I might be interested in making this a reusable function and fixing throughout the app. As I assume it is happening everywhere where that exists.

Slightly wish it could be fixed with valueAsNumber since they have it built in but I cannot find any solutions there that make sense.

**Comment by navDhammu** [comment]
@Duncan-Brain Thanks for the review! 

Yes I think this might be happening in other places where numbers are meant to be edited. But oddly this issue doesn't happen when there are no validation rules on the input. So If you remove the required and the min value and only leave the valueAsNumber in the register options then it works as expected. 

As a side note - I tested the NumberInput I worked on in the other ticket and it doesn't have this issue. So if we're planning on migrating to that then it's safe to say this problem would also be fixed along with it.

**Comment by antsgar** [comment]
@navDhammu thank you for fixing this!

**Review by antsgar — APPROVED** [review]

**Pull request merged**